webknossos/webknossos/client/_resolve_short_link.py

```diff
@@ -0,0 +1,25 @@
+import re
+
+from webknossos.client._generated.api.default import short_link_by_key
+from webknossos.client.context import _get_generated_client, webknossos_context
+
+_SHORT_LINK_REGEX = re.compile(
+    r"^(?P<webknossos_url>https?://.*)/links/" + r"(?P<short_link_key>.*)"
+)
+
+
+def resolve_short_link(url: str) -> str:
+    match = re.match(_SHORT_LINK_REGEX, url)
+    if match is None:
+        return url
+    else:
```

Comment:
Good ideas, thanks, will do both (try-except + urlparse)

Reply:
@philippotto @fm3 Just pushed a version using urlparse and a large try-except. I'll also do a follow-up PR to use urlparse with the annotation and dataset url parsing as well, but to speed things up here I left it out of this PR.

Reply:
Looking good :) I don’t feel super confident approving this PR since there are many line changes in code I’m not familiar with. @philippotto if you feel more confident, maybe you could do the final review?